src/arithmetic/arithmetic_expression.c

```diff
@@ -96,6 +102,31 @@ AR_ExpNode *AR_EXP_NewOpNode(const char *func_name, uint child_count) {
 	return node;
 }
 
+AR_ExpNode *AR_EXP_NewDistinctOpNode(const char *func_name, uint child_count) {
+	assert(func_name);
+
+	AR_ExpNode *node = rm_calloc(1, sizeof(AR_ExpNode));
+	node->type = AR_EXP_OP;
+	node->op.distinct = true;
+	node->op.func_name = func_name;
+	node->op.child_count = child_count;
+	node->op.children = rm_malloc(child_count * sizeof(AR_ExpNode *));
+
+	char *aggregated_func_name;
+	asprintf(&aggregated_func_name, "%sDistinct", func_name);
+
+	AggCtx *agg_func;
+	Agg_GetFunc(aggregated_func_name, &agg_func);
+	free(aggregated_func_name);
+
+	/* TODO: handle Unknown function. */
+	assert(agg_func != NULL);
```

Comment:
We'll fail here on calls like:
`MATCH (a) RETURN SUM(DISTINCT a.age)`

I don't think that the `aggregated_func_name` logic (or more generally, having `AR_EXP_NewDistinctOpNode` as an `AR_EXP_NewOpNode` variant) is a great approach.

If for the moment we're only going to support `COUNT...DISTINCT` and `COLLECT...DISTINCT`, we can name them with string literals on creation, and then don't need to differentiate this way or worry about memory leaks. If, alternately, we want to introduce broad support for distinct aggregations, we should build generic uniquing logic into the aggregate code.